Makefile

```diff
@@ -17,8 +17,12 @@ MAKE_PID := $(shell echo $$PPID)
 JOB_FLAG := $(filter -j%, $(subst -j ,-j,$(shell ps T | grep "^\s*$(MAKE_PID).*$(MAKE)")))
 JOBS := $(subst -j,,$(JOB_FLAG))
 
-release-targets: CPP_FLAGS := -O3 -Wall -c -fmessage-length=0 -fPIC -flto
-test-targets: CPP_FLAGS = -O0 -Wall -c -fmessage-length=0 -fPIC $(GCOV_FLAGS)
+# keep debug option only
+CPP_FLAGS := $(if $(findstring -g,$(CXXFLAGS)), -g)
+LD_FLAGS := $(if $(findstring -g,$(CXXFLAGS)), -g)
```

Comment:
`CXXFLAGS` shouldn't be used for `LD_FLAGS`; the two are for different purposes.

Reply:
Removed, thanks!